.github/workflows/build_library.yml

```diff
@@ -10,31 +10,54 @@ env:
 
 jobs:
   linux:
+    timeout-minutes: 30
     strategy:
       fail-fast: false
       matrix:
         build-arch:
-          - manylinux2010_i686
-          - manylinux2010_x86_64
-          - manylinux2014_aarch64
+          - { image: 'manylinux2014_x86_64', target: 'i686' }
+          - { image: 'manylinux2014_x86_64', target: 'x86_64' }
+          - { image: 'manylinux2014_x86_64', target: 'aarch64' }
 
-    name: Python Linux ${{ matrix.build-arch }}
+    name: Python Linux ${{ matrix.build-arch.target }}
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v3
         with:
           submodules: recursive
 
-      - if: matrix.build-arch == 'manylinux2014_aarch64'
+      - if: matrix.build-arch.target == 'aarch64'
         uses: docker/setup-qemu-action@v1
         with:
           platforms: arm64
 
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Build and export
+        uses: docker/build-push-action@v3
+        with:
+          context: py/
```

Comment:
Does this need context? I wasn't able to see why the build container depends on context -- shouldn't it just bundle build dependencies independent of the application code (same as with the `relay-deps` builder of the main binary)?

Please consider that we implicitly bump the MSRV with every stable release, so it might be simpler if we just forgo any caching here. The library is released so rarely that an additional 1-2 minutes does not matter.